dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -156,6 +157,18 @@ MergeTreeData::MergeTreeData(
         min_format_version = MERGE_TREE_DATA_MIN_FORMAT_VERSION_WITH_CUSTOM_PARTITIONING;
     }
 
+    for (const auto & [name, ast] : getColumns().ttl_expressions)
+    {
+        ASTPtr ttl_ast = ast;
+        auto syntax_result = SyntaxAnalyzer(global_context).analyze(ttl_ast, getColumns().getAllPhysical());
```

Comment:
Maybe this is not good place for this code. TTL doesn't seem to be specific for MergeTree engines only. Maybe add some virtual method for IStorage?